### PR TITLE
EvaluatorUtils to handle label column of type key without text key values

### LINKED ISSUE #395: EvaluatorUtils should handle key label columns with non-text key values
When CV needs to concatenate the per-instance data views, it needs to reconcile the key values from the different folds, even if the key values are not text.

### PR DESCRIPTION
**Pull request by yaeldekel**
If the label column is a key converted from something other than text, 
Fixes #395

### CODE REVIEW

File: src/Microsoft.ML.Data/Evaluators/EvaluatorUtils.cs [hunk 1]
```diff
@@ -595,17 +597,17 @@ public static void ReconcileSlotNames<T>(IHostEnvironment env, IDataView[] views
...
 
             var dvCount = views.Length;
 
-            Dictionary<DvText, int> keyNames;
-            int[] indices;
             // Create mappings from the original key types to the reconciled key type.
-            var keyValueMappers = MapKeys(views.Select(view => view.Schema).ToArray(), columnName, false, out indices, out keyNames);
+            var indices = new int[dvCount];
+            var keyNames = new Dictionary<DvText, int>();
+            var keyValueMappers = Utils.MarshalInvoke(MapKeys<int>, keyValueType.RawType, views.Select(view => view.Schema).ToArray(), columnName, false, indices, keyNames);
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.Data/Evaluators/EvaluatorUtils.cs]: >Utils.MarshalInvoke( [](start = 34, length = 20)

why you need MarshalInvoke?
Can't you just call MapKeys<int>(...) directly?

Comment (Ivanidzo4ka) [inline src/Microsoft.ML.Data/Evaluators/EvaluatorUtils.cs]: >Utils.MarshalInvoke [](start = 34, length = 19)

can you add comment what you use MarshalInvoke to pass keyValueType.RawType as generic type to function?

**Comment by shauheen** [comment]
Thanks @yaeldekel , before merging this PR can you please address questions on the issue this PR is trying to address, and then explain how you are fixing it in this PR.

**Comment by shauheen** [comment]
Thanks @yaeldekel

**Pull request merged**